gis/grass/Portfile

```
@@ -1,11 +1,300 @@
-# -*- coding: utf-8; mode: tcl; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=2:ts=2:sts=2
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup           wxWidgets 1.0
+PortGroup           github 1.0
+PortGroup           debug 1.0
 
+github.setup        OSGeo grass 8.0.0
 name                grass
-epoch               1
-version             6.4.6
-revision            5
-replaced_by         grass7
-PortGroup           obsolete 1.0
+set main_version    [join [lrange [split ${version} "."] 0 1] ""]
+revision            0
+set realVersion     ${version}
+distname            grass-${realVersion}
+maintainers         nomaintainer
+categories          gis
+license             GPL-2+
+platforms           darwin
 
+description         GRASS
+long_description    GRASS is a Geographic Information System (GIS) used for \
+                    geospatial data management and analysis.
+
+homepage            http://grass.osgeo.org/
+master_sites        ${homepage}grass[join [lrange [split ${realVersion} .] 0 1] {}]/source/
+
+checksums           rmd160  2192eb3ef991dd9645a7521f691aa9dc12bc4f16 \
+                    sha256  43231ae62874ace92b35d1457950749046aae374e4d9af9d9da22cf9c55d22a1 \
+                    size    64944539
+
+depends_build       port:pkgconfig
+depends_lib         port:bzip2 \
+                    port:freetype \
+                    port:fftw-3 \
+                    port:gdal \
+                    port:pdal \
+                    port:geos \
+                    port:tiff \
+                    port:libpng \
+                    port:liblas\
+                    path:lib/pkgconfig/cairo.pc:cairo \
+                    port:zlib \
+                    port:readline
+
+patchfiles          800-patch.diff
+
+configure.cflags    -DGL_SILENCE_DEPRECATION
+configure.ldflags   -Wl,-rpath,${prefix}/lib
```

You normally should not overwrite the flags MacPorts provides, but you can append to them.
```suggestion
configure.cflags-append -DGL_SILENCE_DEPRECATION
configure.ldflags-append -Wl,-rpath,${prefix}/lib
```
Why is rpath needed? Usually we try to avoid rpath in MacPorts.

---

The configure settings for shared library linking are hard coded per platform [(aclocal.m4)](https://github.com/OSGeo/grass/blob/eba61a1ca122758c9aae4ff0c28bab895609d9f6/aclocal.m4#L506-L512). The change to rpath was made for the official dist of the GRASS.app bundle.
Temporary solution is a provided  patch. I have planned to make these settings at least overridable during configuration.